### PR TITLE
Add metrics library to provide health check endpoint

### PR DESCRIPTION
**Pull request by purple52**

### CODE REVIEW

File: metrics/src/main/java/metrics/MetricsServletContextListener.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.servlets.MetricsServlet;
+
+import javax.servlet.annotation.WebListener;
+
+/**
+ * Annotated context listener for wiring up the metrics servlet.
+ */
+@WebListener
+public class MetricsServletContextListener extends MetricsServlet.ContextListener {
```
Comment (matt-rich) [inline metrics/src/main/java/metrics/MetricsServletContextListener.java]: Should this be an abstract class? Or is it just part of how the drop wizard pulls it in?

File: metrics/src/main/java/metrics/HealthCheckServletContextListener.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.codahale.metrics.servlets.HealthCheckServlet;
+
+import javax.servlet.annotation.WebListener;
+
+/**
+ * Annotated context listener for wiring up the health check servlet.
+ */
+@WebListener
+public class HealthCheckServletContextListener extends HealthCheckServlet.ContextListener {
```
Comment (matt-rich) [inline metrics/src/main/java/metrics/HealthCheckServletContextListener.java]: static import for HealthCheckServlet.ContextListerner

Comment (matt-rich) [inline metrics/src/main/java/metrics/HealthCheckServletContextListener.java]: Should this be an abstract class with static getter? Or is this just how it has to be to work with the framework?

File: metrics/src/main/java/metrics/MetricsServletContextListener.java (hunk 1 above)
Comment (purple52) [inline metrics/src/main/java/metrics/MetricsServletContextListener.java]: This needs to implement the abstract non-static getter required by the abstract ContextListener Dropwizard class.

File: metrics/src/main/java/metrics/HealthCheckServletContextListener.java (hunk 2 above)
Comment (purple52) [inline metrics/src/main/java/metrics/HealthCheckServletContextListener.java]: This needs to implement the abstract non-static getter required by the abstract ContextListener Dropwizard class.

**Pull request merged**